File: JavaScript/Max Sum Contiguous Subarray.js
module.exports = {
  //param A : array of integers
  //return an integer
  maxSubArray: function (A) {
    maxSum = A[0];
    sum = 0;

    for (i = 0; i < A.length; i++) {
      sum = Math.max(sum + A[i], A[i]);
      maxSum = Math.max(sum, maxSum);
    }

    return maxSum;
  },
};

File: Python/Max Sum Contiguous Subarray.py
class Solution:
    # @param A : tuple of integers
    # @return an integer
    def maxSubArray(self, A):
        maxSum=A[0]
        sum=0

        for i in A:
            sum = max (sum+i,i)
            maxSum = max(sum,maxSum)

        return maxSum
